test/functional/test_framework/test_node.py

```diff
@@ -98,14 +99,23 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-uacomment=testnode%d" % i,
         ]
         if use_valgrind:
+            valgrind_version = self.get_valgrind_version()
+            if not valgrind_version:
+                raise Exception('Valgrind not present')
+            exit_on_first_error_flag = ""
+            if valgrind_version and valgrind_version >= MIN_VALGRIND_VERSION_EARLY_EXIT:
```

Comment:
This can be reduced to `if valgrind_version >= MIN_VALGRIND_VERSION_EARLY_EXIT:` since the `not valgrind_version` case is handled on L103.